ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -273,6 +273,11 @@ func New(ctx context.Context, options ...Option) (Client, error) {
 			return Client{}, err
 		}
 	}
+	// Wrap RPC client to have more contextualized errors
+	c.RPC = rpcWrapper{
```

Comment:
Should we make rpcWrapper a public struct? In general I think that it's a good practice to use public struct types if they're being assigned to a public variable.

Because user may want to create their own struct or init a variable (except `:=`) with a field inside that has the same type, but they cannot do it if the struct type is private.

Reply:
```go
// Client is a client to access your chain by querying and broadcasting transactions.
type Client struct {
	// RPC is Tendermint RPC.
	RPC rpcclient.Client
        // ...
}
```

`client.RPC` is now an interface, so the user can assign it a variable of this same interface type :

```go
import (
    rpcclient "github.com/tendermint/tendermint/rpc/client"
    "github.com/ignite/cli/ignite/pkg/cosmosclient"
 )

var rpc rpcclient.Client = cosmosclient.Client{}.RPC
```

Reply:
Oh, got it.

Reply:
To add on this, I think there's sufficient number of types in the cosmos-sdk so we don't add our own :)